### PR TITLE
feat: move all environment methods to plain client

### PR DESCRIPTION
**Pull request by suevalov**
Exposes all environment-related endpoints to a plain CMA client and reuses endpoints in the original CMA client internals.

### CODE REVIEW

File: lib/plain/endpoints/environment.ts [hunk 1]
```diff
@@ -13,6 +14,10 @@ export const get = (http: AxiosInstance, params: GetEnvironmentParams) => {
   )
 }
 
+export const getAll = (http: AxiosInstance, params: GetSpaceParams) => {
```
Comment (anho) [inline lib/plain/endpoints/environment.ts:17]: That endpoint is paginated, so it should be getMany with additional query params.

File: lib/entities/environment.ts [hunk 2]
```diff
@@ -22,6 +22,10 @@ export type EnvironmentProps = {
   name: string
 }
 
+export type CreateEnvironmentProps = {
```
Comment (anho) [inline lib/entities/environment.ts:25]: What about `type CreateEnvironmentProps  = Partial<Omit<EnvironmentProps, 'sys'>>`?

File: lib/create-space-api.ts [hunk 3]
```diff
@@ -184,18 +195,18 @@ export default function createSpaceApi({ http }: { http: AxiosInstance }) {
...
-      sourceEnvironmentId?: string
-    ) {
-      return http
-        .put('environments/' + id, data, {
-          headers: sourceEnvironmentId
-            ? { 'X-Contentful-Source-Environment': sourceEnvironmentId }
-            : {},
-        })
-        .then((response) => wrapEnvironment(http, response.data), errorHandler)
+    createEnvironmentWithId(id: string, data: CreateApiKeyProps, sourceEnvironmentId: string) {
```
Comment (anho) [inline lib/create-space-api.ts:198]: The type of data looks wront.

File: lib/plain/endpoints/environment.ts (hunk 1 above)
Comment (suevalov) [reply lib/plain/endpoints/environment.ts:17]: it's not paginated according to the documentation and the current implementation of the main client.

Comment (anho) [reply lib/plain/endpoints/environment.ts:17]: Interestingly, the API does answer with a paginated request and I am able to utilise skip and limit.

File: lib/create-space-api.ts (hunk 3 above)
Comment (suevalov) [reply lib/create-space-api.ts]: fixed

File: lib/plain/endpoints/environment.ts (hunk 1 above)
Comment (suevalov) [reply lib/plain/endpoints/environment.ts:17]: pricing limits are always below 100 envs, so all our API clients expose `getAll` instead of `getMany`

**Pull request merged**